src/set_theory/ordinal/arithmetic.lean

```diff
@@ -877,24 +877,18 @@ theorem dvd_add_iff : ∀ {a b c : ordinal}, a ∣ b → (a ∣ b + c ↔ a ∣
 theorem dvd_add {a b c : ordinal} (h₁ : a ∣ b) : a ∣ c → a ∣ b + c :=
 (dvd_add_iff h₁).2
 
-theorem dvd_zero (a : ordinal) : a ∣ 0 := ⟨_, (mul_zero _).symm⟩
-
-theorem zero_dvd {a : ordinal} : 0 ∣ a ↔ a = 0 :=
-⟨λ ⟨h, e⟩, by simp only [e, zero_mul], λ e, e.symm ▸ dvd_zero _⟩
-
-theorem one_dvd (a : ordinal) : 1 ∣ a := ⟨a, (one_mul _).symm⟩
-
 theorem div_mul_cancel : ∀ {a b : ordinal}, a ≠ 0 → a ∣ b → a * (b / a) = b
 | a _ a0 ⟨b, rfl⟩ := by rw [mul_div_cancel _ a0]
 
 theorem le_of_dvd : ∀ {a b : ordinal}, b ≠ 0 → a ∣ b → a ≤ b
 | a _ b0 ⟨b, rfl⟩ := by simpa only [mul_one] using mul_le_mul_left'
   (one_le_iff_ne_zero.2 (λ h : b = 0, by simpa only [h, mul_zero] using b0)) a
 
-theorem dvd_antisymm {a b : ordinal} (h₁ : a ∣ b) (h₂ : b ∣ a) : a = b :=
-if a0 : a = 0 then by subst a; exact (zero_dvd.1 h₁).symm else
-if b0 : b = 0 then by subst b; exact zero_dvd.1 h₂ else
-le_antisymm (le_of_dvd b0 h₁) (le_of_dvd a0 h₂)
+instance : is_antisymm ordinal (∣) :=
```

Comment:
Isn't this a general thing about linear ordered semirings? It's also true for `ℕ`.

Reply:
I think it is, and I think we're also somehow missing this theorem. I'll split this PR into two: one adding the missing results on `dvd`, and one that removes the redundant ones from `ordinal`.

Reply:
Actually, this wouldn't actually help in the case of ordinals, since they aren't a linear ordered semiring (not left-distributive). Is there perhaps a weaker typeclass that could help here?

Reply:
Here's an interesting question: what typeclass guarantees this theorem?
```lean
theorem le_of_dvd {a b : α} (h : 0 < b) (hab : a ∣ b) : a ≤ b := sorry
```

Reply:
The correct thing is `canonically_ordered_monoid_with_zero`, but it doesn't exist.

Reply:
That's unfortunate. I guess all of this means this PR can just stay as is?

Reply:
Yeah, we already have `nat.le_of_dvd`, `int.le_of_dvd` and `pnat.le_of_dvd`.